src/proto/generation.proto

```diff
@@ -240,6 +241,92 @@ message ClassifierParameters {
   optional Action realized_action = 3;
 }
 
+//
+// Transforms
+//
+
+enum BorderMode {
+  BORDER_REFLECT = 0;   // reflect image values across the border
+  BORDER_REPLICATE = 1; // replicate border values outside the image
+  BORDER_WRAP = 2;      // wrap around / tile the image values
+  BORDER_ZERO = 3;      // use 0 for locations outside the image
+}
+
+enum ColorMatchMode {
+  COLOR_MATCH_HSV = 0;  // match hue, saturation, and value histograms
+  COLOR_MATCH_LAB = 1;  // match lightness, a, and b histograms
+  COLOR_MATCH_RGB = 2;  // match red, green, and blue histograms
+}
+
+message TransformAddNoise {
+  float amount = 1; // amount of noise to add
+  uint32 seed = 2;  // random seed for noise generation
+}
+
+message TransformBlend {
+  float amount = 1;     // blend amount from 0 to 1
+  Artifact target = 2;  // target image
+}
+
+message TransformColorMatch {
+  ColorMatchMode color_mode = 1;
+  Artifact image = 2;
+}
+
+message TransformContrast {
+  optional float brightness = 1;  // scale on pixel intensities, 1.0 is no change
+  optional float contrast = 2;    // contrast adjustment, 1.0 is no change
+}
+
+message TransformDepthCalc {
+  optional bool export = 1;        // return the depth map as an artifact
```

Comment:
Do we have the depth map defined?

Reply:
When requested it is returned as PNG image artifact for use in post.

Reply:
You should not be baking in assumptions that it will be a PNG.